#!/usr/bin/env python
PACKAGE = "image_cloud"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("radius",    double_t,   0, "The radius for normal estimation and non maxima supression", 0.2,  0, 4)
gen.add("radius_search",    double_t,   0, "Sphere radius that is to be used for determining the nearest neighbors used for the key point detection", 0.1,  0, 5)
gen.add("threshold",    double_t,   0, "the threshold value for detecting corners. This is only evaluated if non maxima suppression is turned on.", 0.0,  0, 5)
gen.add("non_max_supression",    bool_t,   0, "Whether non maxima suppression should be applied or the response for each point should be returned, this value needs to be turned on in order to apply thresholding and refinement",  True)
gen.add("refine", bool_t, 0, " Whether the detected key points should be refined or not. If turned of, the key points are a subset of the original point cloud. Otherwise the key points may be arbitrary.",True)
gen.add("subscribe_topic",   str_t,   0, "Subscribe Topic",  "")
gen.add("publish_topic",     str_t,   0, "Publish Topic",  "")

exit(gen.generate(PACKAGE, "image_clouud", "harris3d"))
